assets/src/components/parts/janus-hub-spoke/HubSpoke.tsx

```diff
@@ -0,0 +1,378 @@
+    if (typeof props?.model === 'string') {
+      try {
+        pModel = JSON.parse(props.model);
+        setModel(pModel);
+      } catch (err) {
+        // bad json, what do?
+      }
+    }
+    if (!pModel) {
+      return;
+    }
+    initialize(pModel);
+  }, [props]);
+
+  useEffect(() => {
+    if (!ready) {
+      return;
+    }
+    props.onReady({ id, responses: [] });
+  }, [ready]);
+
+  const handleButtonPress = (val: any) => {
+    props.onSubmit({
+      id: `${id}`,
+      responses: [
+        {
+          key: 'selectedSpoke',
+          type: CapiVariableTypes.NUMBER,
+          value: val,
+        },
+      ],
+    });
+  };
+  useEffect(() => {
+    if (!props.notify) {
+      return;
+    }
+    const notificationsHandled = [
+      NotificationType.CHECK_STARTED,
+      NotificationType.CHECK_COMPLETE,
+      NotificationType.CONTEXT_CHANGED,
+      NotificationType.STATE_CHANGED,
+    ];
+    const notifications = notificationsHandled.map((notificationType: NotificationType) => {
+      const handler = (payload: any) => {
+        /* console.log(`${notificationType.toString()} notification handled [MCQ]`, payload); */
+        switch (notificationType) {
+          case NotificationType.CHECK_STARTED:
+            // should disable input during check?
+            break;
+          case NotificationType.CHECK_COMPLETE:
+            // if disabled above then re-enable now
+            break;
+          case NotificationType.STATE_CHANGED:
+            {
+              //
+            }
+            break;
+          case NotificationType.CONTEXT_CHANGED:
+            {
+              if (payload.mode === contexts.REVIEW) {
+                setEnabled(false);
+              }
+            }
+            break;
+        }
+      };
+      const unsub = subscribeToNotification(props.notify, notificationType, handler);
+      return unsub;
+    });
+    return () => {
+      notifications.forEach((unsub) => {
+        unsub();
+      });
+    };
+  }, [props.notify, options, handleButtonPress]);
+
+  // Set up the styles
+  const styles: CSSProperties = {
+    width,
+  };
+  if (overrideHeight) {
+    styles.height = height;
+    styles.marginTop = '8px';
+  }
+
+  useEffect(() => {
+    const styleChanges: any = {};
+    if (width !== undefined) {
+      styleChanges.width = { value: width as number };
+    }
+    if (height != undefined) {
+      styleChanges.height = { value: height as number };
+    }
+
+    props.onResize({ id: `${id}`, settings: styleChanges });
+  }, [width, height]);
+
+  let columns = 1;
+  if (customCssClass === 'two-columns') {
+    columns = 2;
+  }
+  if (customCssClass === 'three-columns') {
+    columns = 3;
+  }
+  if (customCssClass === 'four-columns') {
+    columns = 4;
+  }
+  const progressPercentage = options?.length ? (completedSpokeCount / options?.length) * 100 : 0;
+
+  return (
+    <React.Fragment>
+      {ready ? (
+        <>
+          <style>
+            {`
+
```

Comment:
I'm no expert in the advanced author space, but directly embedding styles like this seems like an anti-pattern. Should we not be defining these styles in a css file?

Also, using `!important` is a pretty heavy hammer that should really only be used when necessary. Most of the time when it is used, the same result can be achieved by properly following CSS specificity and precedence https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_cascade/Specificity